src/plugins/intel_gna/src/transformations/gather_sinking_fuse.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2023 Intel Corporation2
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+
+/**
+ * @brief Fuses (merges) 2 consecutive Gather nodes into one. If resulting Gather node
+ * does nothing with input (it is not actually permute items), just removes Gather nodes.
+ *
+ *   Any #1                  Any #1
+ *     |                        |
+ *   Gather #1              Gather #3
+ *     |             =>         |
+ *   Gather #2                Any #2
+ *     |
+ *    Any #2
+ *
+ *  or
+ *
+ *    Any #1                  Any #1
+ *     |                        |
+ *   Gather #1                Any #2
+ *     |             =>
```

Comment:
What about Gather-Reshape-Gather? I think they can be also merged

Reply:
Discussed on the meeting. We will create separate PR for it